app/routes/users/components/PenaltyForm.js

```diff
@@ -0,0 +1,142 @@
+// @flow
+
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import cx from 'classnames'; //???
+import React, { Component } from 'react';
+import Button from 'app/components/Button';
+import { Form, TextArea, TextInput, SelectInput } from 'app/components/Form';
+import { reduxForm, Field, reset } from 'redux-form';
+import { addPenalty } from 'app/actions/UserActions';
+import type { AddPenalty, ID } from 'app/models';
+
+type Props = {
+  placeholder?: string,
+  user: ID,
+  addPenalty: AddPenalty => Promise<*>,
+  reason?: string,
+  weight?: number,
+  handleSubmit: Function => void,
+  actionGrant: boolean,
+  hidden?: boolean,
+  button?: boolean,
+  className?: string
+};
+
+type State = {
+  hidden: boolean,
+  sent: boolean
+};
+
+class AnnouncementInLine extends Component<Props, State> {
+  props: Props;
+  state = {
+    hidden: true,
+    sent: false
+  };
+
+  onSubmit = (penalty, user) => {
+    //?
+    this.props.addPenalty({
+      ...penalty,
+      sourceEvent: penalty.sourceEvent.value,
+      user
+    });
+    this.setState(() => ({
+      sent: true
+    }));
+  };
+
+  handleHide = () => {
+    this.setState(prevState => ({
+      hidden: !prevState.hidden
+    }));
+  };
+
+  render() {
+    const {
+      actionGrant,
+      handleSubmit,
+      placeholder,
```

Comment:
You can remove this `placeholder` functionality.

Reply:
Placeholder is used in the different fields in the form, should I replace them with another function?